src/bootstrap/build/mod.rs

```diff
@@ -501,6 +501,18 @@ impl Build {
         if target != self.config.build && !target.contains("msvc") {
             base.push(format!("-Clinker={}", self.cc(target).display()));
         }
+        // If we have an external LLVM ensure that crates can link to it
+        let external_llvm = self.config.target_config.get(target).and_then(|config| {
+            config.llvm_config.clone()
+        });
+        if let Some(mut llvm_config) = external_llvm {
+            llvm_config.pop();
+            llvm_config.pop();
+            // Search in <llvm-root>/bin for shared libraries on Windows
+            base.push(format!("-L{}", llvm_config.join("bin").to_str().unwrap()));
+            // Search in <llvm-root>/lib elsewhere
+            base.push(format!("-L{}", llvm_config.join("lib").to_str().unwrap()));
```

Comment:
The problem is that `rustc_llvm` is in the sysroot, so its dependencies are not handled by cargo.

See https://github.com/rust-lang/rust/issues/30970